### PR TITLE
Ca api

### PR DESCRIPTION
**Pull request by CallumD**
Was able to just switch to the new basic authenticator. 

@raphael can you please review. Please be brutal as 

I have tested this against cobalt-403 (ca1), moo (moo-analytics) and my (analytics) and all working well!!!!

@alikhajeh1 I know you will also be interested in this.

### CODE REVIEW

File: ca/commands.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import (
+	"net/http"
+
+	"github.com/rightscale/rsc/rsapi"
+)
+
+const (
+	// Used by rsc to display command line help
+	ApiName = "RightScale Cloud Ananlytics APIs"
```
Comment (CallumD) [inline ca/commands.go]: typo

File: ca/ca.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	"github.com/rightscale/rsc/cmd"
+	"github.com/rightscale/rsc/metadata"
+	"github.com/rightscale/rsc/rsapi"
+	"regexp"
+)
+
+// Metadata synthetized from all CA APIs metadata
+var GenMetadata map[string]*metadata.Resource
+
+// CA 1.0 common client to all self-service APIs
```
Comment (raphael) [inline ca/ca.go]: client to self-service...

File: ca/ca.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	api.Host = apiHostFromLogin(cmdLine.Host)
+	api.Metadata = GenMetadata
+	return &Api{api}, nil
+}
+
+func apiHostFromLogin(host string) string {
+	integration, _ := regexp.MatchString("^cobalt", host)
+	staging, _ := regexp.MatchString("^moo", host)
+
+	switch {
```
Comment (raphael) [inline ca/ca.go]: Is it possible to write generic code like the one for the self-service host instead? (capture the relevant prefix and suffix using a regexp)

**Comment by alikhajeh1** [comment]
@callumD nice one! I look forward to playing with this.

**Comment by CallumD** [comment]
Code fixed up @raphael can you please re review. If you are happy I will squash and merge.

**Comment by raphael** [comment]
Looks good

**Comment by CallumD** [comment]
@raphael I have squashed but I have no idea how we release this. Is it safe for me to merge?

**Pull request merged**